### PR TITLE
chore: bump celestia app v1.0.0-rc9

### LINKED ISSUE #2448: Bump rsmt2d to 0.10.0
https://github.com/celestiaorg/rsmt2d/releases/tag/v0.10.0

### PR DESCRIPTION
**Pull request by evan-forbes**
## Overview

this PR bumps app, core, the sdk, and rsmt2d

besides the update to rsmt2d, the only app/core related changes are to the timeout commit. We're reverting to what was used for previous testnets (such as BSR), where we have a static timeout after reaching 2/3s in the commit instead of a dynamic one.

closes #2448 
## Checklist

- [x] New and updated code has appropriate documentation
- [x] New and updated code has new and/or updated testing
- [x] Required CI checks are passing
- [x] Visual proof for any user facing features like CLI or documentation updates
- [ ] Linked issues closed with keywords

### CODE REVIEW

**Comment by evan-forbes** [comment]
leaving as a draft until the official tag comes out

also `TestDASer_stopsAfter_BEFP` is failing which I think makes sense since we've updated rsmt2d, but I'm not entirely sure what's the best way to fix it.

File: share/eds/retriever.go [hunk 1]
```diff
@@ -286,7 +286,10 @@ func (rs *retrievalSession) doRequest(ctx context.Context, q *quadrant) {
 				if rs.square.GetCell(uint(x), uint(y)) != nil {
 					return
 				}
-				rs.square.SetCell(uint(x), uint(y), share)
+				if err := rs.square.SetCell(uint(x), uint(y), share); err != nil {
+					// TODO: handle error
+					return
+				}
```
Comment (evan-forbes) [inline share/eds/retriever.go:292]: note for reviewers

**Comment by evan-forbes** [comment]
anyone should feel free to push to this PR to fix or update things as I might be gone cc @rootulp @Wondertan

File: go.mod [hunk 2]
```diff
@@ -10,12 +10,12 @@ require (
 	github.com/BurntSushi/toml v1.3.2
 	github.com/alecthomas/jsonschema v0.0.0-20200530073317-71f438968921
 	github.com/benbjohnson/clock v1.3.5
-	github.com/celestiaorg/celestia-app v1.0.0-rc8
+	github.com/celestiaorg/celestia-app v1.0.0-rc8.0.20230711195230-7a176b2bb786 // v1.0.0-rc8
```
Comment (evan-forbes) [inline go.mod:13]: we can bump to an official tag after https://github.com/celestiaorg/celestia-app/pull/2081

Comment (rootulp) [reply go.mod:13]: Created https://github.com/celestiaorg/celestia-app/releases/tag/v1.0.0-rc9 I just can't push to this branch b/c permissions issue 😢 

```
~/git/rootulp/celestia/celestia-node on evan/bump-ce..p-v1.0.0-rc9 >1

$ git push
ERROR: Permission to celestiaorg/celestia-node.git denied to rootulp.
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
```

[the remark this answers is not in the document]
Comment (rootulp) [reply go.mod]: Resolved via https://github.com/celestiaorg/celestia-node/pull/2458/commits/d04618112569487d4437819af08254a6719676fb

Comment (rootulp) [reply go.mod]: Resolved via https://github.com/celestiaorg/celestia-node/pull/2458/commits/d04618112569487d4437819af08254a6719676fb

File: share/eds/retriever.go (hunk 1 above)
Comment (Wondertan) [inline share/eds/retriever.go:292]: As SetCell now internally checks if the Cell was written before, we can remove the GetCell call above(which does exactly this check). The error returned from SetCell can be safely ignored.

File: nodebuilder/tests/swamp/config.go [hunk 3]
```diff
@@ -16,7 +16,7 @@ type Config struct {
 func DefaultConfig() *Config {
 	cfg := core.DefaultTestConfig()
 	// target height duration lower than this tend to be flakier
```
Comment (rootulp) [inline nodebuilder/tests/swamp/config.go:18]:
```suggestion
	// timeout commit lower than this tend to be flakier
```

Comment (rootulp) [reply nodebuilder/tests/swamp/config.go:18]: [optional] we may not even need to override this value if the default is 300 milliseconds: https://github.com/rootulp/celestia-app/blob/0d93ec79bc65f8a2bb37c4bb8a1fb7140a033251/test/util/testnode/config.go#L139-L142

^ needs confirmation by running this test without lines 18 - 19

File: share/eds/eds.go [hunk 4]
```diff
@@ -231,8 +234,18 @@ func prependNamespace(quadrant int, shr share.Share) []byte {
...
+		return nil, err
+	}
+	colRoots, err := eds.ColRoots()
+	if err != nil {
+		return nil, err
+	}
+
+	roots := make([][]byte, 0, len(rowRoots)+len(colRoots))
+	roots = append(roots, rowRoots...)
+	roots = append(roots, colRoots...)
```
Comment (rootulp) [inline share/eds/eds.go:248]: Motivated https://github.com/celestiaorg/rsmt2d/issues/249

File: share/eds/eds_test.go [hunk 5]
```diff
@@ -167,14 +167,26 @@ func TestInnerNodeBatchSize(t *testing.T) {
...
+	loadedRowRoot, err := loaded.RowRoots()
+	require.NoError(t, err)
+	require.Equal(t, rowRoot, loadedRowRoot)
+
+	colRoot, err := eds.ColRoots()
+	require.NoError(t, err)
+	loadedColRoot, err := loaded.ColRoots()
+	require.NoError(t, err)
+
+	require.Equal(t, colRoot, loadedColRoot)
```
Comment (rootulp) [inline share/eds/eds_test.go:189]: [nits] pluralize

- `rowRoot` => `rowRoots`
-  `loadedRowRoot` => `loadedRowRoots`
- `colRoot` => `colRoots`
- `loadedColRoot` => `loadedColRoots`

**Review by rootulp — APPROVED** [review]

File: share/eds/eds_test.go (hunk 5 above)
Comment (rootulp) [reply share/eds/eds_test.go]: Resolved via 51c6e14907f858a00fe5b49a934afb66a0482c07

**Review by rootulp — APPROVED** [review]

**Review by Wondertan — APPROVED** [review]
LGTM and needs one more review from the node team member

**Review by walldiss — APPROVED** [review]

**Pull request merged**